src/component_macro_reference.md

```diff
@@ -0,0 +1,228 @@
+# The `component!` macro reference
```

Comment:
Technically, the exclamation mark is only used for function-like macros, so I would just call it he component macro

Reply:
Nice catch. Fixed.